### PR TITLE
chore: add auto-approve bot

### PR DESCRIPTION
**Pull request by parthea**
Adds an [auto approve configuration](https://github.com/googleapis/repo-automation-bots/blob/master/packages/auto-approve/README.md) to automatically approve the PRs that update discovery artifacts.

Only PRs that match these conditions will be approved:
- PR Author is `yoshi-code-bot`
- PR Title is `chore: Update discovery artifacts`
- Files changed limited to the following
`docs/dyn/index.md` , `docs/dyn/*.html"` or `googleapiclient/discovery_cache/documents/*.json`

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -8,4 +8,6 @@
 *               @googleapis/yoshi-python
 
 # The python-samples-reviewers team is the default owner for samples changes
-/samples/  @googleapis/python-samples-owners
\ No newline at end of file
+/samples/  @googleapis/python-samples-owners
+
+.github/auto-approve.yml                 @googleapis/github-automation
```
Comment (busunkim96) [inline .github/CODEOWNERS:13]: Should this config be exclusively owned by github-automation? Can we also add yoshi-python?

File: .github/auto-approve.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+rules:
+
+# incoming PRs from yoshi-code-bot
+- author: "yoshi-code-bot"
+
+# Title for PRs from yoshi-code-bot MUST start with `chore: Update discovery artifacts`.
+# Title must be regex.
+  title: "^chore: Update discovery artifacts"
+
+# (Optional) The only changed file paths in an incoming PR MUST be exactly
```
Comment (busunkim96) [inline .github/auto-approve.yml:10]:
```suggestion
# The only changed file paths in an incoming PR MUST be exactly
```

File: .github/auto-approve.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+rules:
```
Comment (busunkim96) [inline .github/auto-approve.yml:1]:
```suggestion
# https://github.com/googleapis/repo-automation-bots/tree/master/packages/auto-approve

rules:
```

Small suggestion to make it easier to find the bot documentation/code.

File: .github/CODEOWNERS (hunk 1 above)
Comment (parthea) [reply .github/CODEOWNERS]: Good catch! Fixed.

**Comment by parthea** [comment]
@busunkim96 PTAL

**Review by busunkim96 — APPROVED** [review]

**Pull request merged**